subprojects/core/src/main/java/org/gradle/api/internal/collections/AbstractIterationOrderRetainingElementSource.java

```diff
@@ -329,11 +330,12 @@ public void realize() {
         }
 
         @Override
-        public void collectInto(ImmutableCollection.Builder<T> builder) {
+        public Value<Void> collectInto(ImmutableCollection.Builder<T> builder) {
```

Comment:
```suggestion
        public void collectInto(ImmutableCollection.Builder<T> builder) {
```

Can you revert to returning `void` here?

Reply:
Done